test/common/k8s/azureassignedidentity/azureassignedidentity.go

```diff
@@ -63,7 +63,9 @@ func WaitOnLengthMatched(target int) (bool, error) {
 				list, err := GetAll()
 				if err != nil {
 					errorChannel <- err
+					return
 				}
+				// len of nil slices is 0, so shouldn't panic here
```

Comment:
The panic here was happening because when there is an error in `GetAll()` we return nil for list. With the return on error that should be resolved.